### PR TITLE
Fixing a cast that was causing in overflow in IndexSegment

### PR DESCRIPTION
**Pull request by vgkholla**

### CODE REVIEW

File: ambry-store/src/main/java/com.github.ambry.store/IndexValue.java [hunk 1]
```diff
@@ -105,11 +105,11 @@
         offset = new Offset(logSegmentName, value.getLong());
         flags = value.get();
         int expiresAt = value.getInt();
-        expiresAtMs = expiresAt != Utils.Infinite_Time ? (expiresAt * Time.MsPerSec) : Utils.Infinite_Time;
+        expiresAtMs = expiresAt != Utils.Infinite_Time ? ((long) expiresAt * Time.MsPerSec) : Utils.Infinite_Time;
```
Comment (pnarayanan) [inline ambry-store/src/main/java/com.github.ambry.store/IndexValue.java]: might as well declare `expiresAt` as a long to begin with?

File: ambry-store/src/test/java/com.github.ambry.store/IndexValueTest.java [hunk 2]
```diff
@@ -60,9 +61,9 @@ public void putValueTest() throws InterruptedException {
     String logSegmentName = LogSegmentNameHelper.getName(pos, gen);
     long size = Utils.getRandomLong(TestUtils.RANDOM, 1000);
     long offset = Utils.getRandomLong(TestUtils.RANDOM, 1000);
-    long expiresAtMs = Utils.getRandomLong(TestUtils.RANDOM, 1000000);
+    long expiresAtMs = Utils.getRandomLong(TestUtils.RANDOM, 1000000) + SystemTime.getInstance().milliseconds();
     long expectedExpirationTimeV1 = Utils.getTimeInMsToTheNearestSec(expiresAtMs);
-    long operationTimeAtMs = Utils.getRandomLong(TestUtils.RANDOM, 1000000);
+    long operationTimeAtMs = Utils.getRandomLong(TestUtils.RANDOM, 1000000) + +SystemTime.getInstance().milliseconds();
```
Comment (pnarayanan) [inline ambry-store/src/test/java/com.github.ambry.store/IndexValueTest.java]: extra `+` here.

File: ambry-store/src/test/java/com.github.ambry.store/IndexValueTest.java [hunk 3]
```diff
@@ -60,9 +61,9 @@ public void putValueTest() throws InterruptedException {
     String logSegmentName = LogSegmentNameHelper.getName(pos, gen);
     long size = Utils.getRandomLong(TestUtils.RANDOM, 1000);
     long offset = Utils.getRandomLong(TestUtils.RANDOM, 1000);
-    long expiresAtMs = Utils.getRandomLong(TestUtils.RANDOM, 1000000);
+    long expiresAtMs = Utils.getRandomLong(TestUtils.RANDOM, 1000000) + SystemTime.getInstance().milliseconds();
```
Comment (pnarayanan) [inline ambry-store/src/test/java/com.github.ambry.store/IndexValueTest.java]: So this test fails without the fix above?

Comment (vgkholla) [inline ambry-store/src/test/java/com.github.ambry.store/IndexValueTest.java]: Yes, that's right.

**Comment by nsivabalan** [comment]
Looks good. Merging.

**Pull request merged**